pkg/iac/scanners/dockerfile/parser/parser.go

```diff
@@ -13,7 +13,34 @@ import (
 	"github.com/aquasecurity/trivy/pkg/iac/providers/dockerfile"
 )
 
-func Parse(_ context.Context, r io.Reader, path string) ([]*dockerfile.Dockerfile, error) {
+type Parser struct {
+	strict bool
+}
+
+type Option func(p *Parser)
+
+// WithStrict returns a Parser option that enables strict parsing mode.
+// For example, unknown flags cause an error when strict is true, otherwise they are ignored.
+func WithStrict(yes bool) Option {
+	return func(p *Parser) {
+		p.strict = yes
+	}
+}
+
+func Parse(ctx context.Context, r io.Reader, path string) ([]*dockerfile.Dockerfile, error) {
+	p := NewParser()
+	return p.Parse(ctx, r, path)
+}
```

Comment:
This looks a bit strange.
Maybe we can execute this code at the call site instead?
```go
p := NewParser()
return p.Parse(ctx, r, path)
```